internal/xcuit/config.go

```diff
@@ -0,0 +1,136 @@
+type Suite struct {
+	Name        string     `yaml:"name,omitempty" json:"name"`
+	Devices     []Device   `yaml:"devices,omitempty" json:"devices"`
+	TestOptions TestOption `yaml:"testOptions,omitempty" json:"testOptions"`
+}
+
+// Device represents device configuration.
+type Device struct {
+	ID              string  `yaml:"id,omitempty" json:"id"`
+	Name            string  `yaml:"name,omitempty" json:"name"`
+	PlatformVersion string  `yaml:"platformVersion,omitempty" json:"platformVersion"`
+	PlatformName    string  `yaml:"platformName,omitempty" json:"platformName"`
+	Options         Options `yaml:"options,omitempty" json:"options"`
+}
+
+// Options represents device options configuration.
+type Options struct {
+	CarrierConnectivity bool   `yaml:"carrierConnectivity,omitempty" json:"carrierConnectivity"`
+	DeviceType          string `yaml:"deviceType,omitempty" json:"deviceType"`
+	Private             bool   `yaml:"private,omitempty" json:"private"`
+}
+
+// FromFile creates a new xcuit Project based on the filepath cfgPath.
+func FromFile(cfgPath string) (Project, error) {
+	var p Project
+
+	f, err := os.Open(cfgPath)
+	if err != nil {
+		return Project{}, fmt.Errorf("failed to locate project config: %v", err)
+	}
+	defer f.Close()
+
+	if err := yaml.NewDecoder(f).Decode(&p); err != nil {
+		return Project{}, fmt.Errorf("failed to parse project config: %v", err)
+	}
+	p.ConfigFilePath = cfgPath
+
+	if p.Sauce.Concurrency < 1 {
+		// Default concurrency is 2
+		p.Sauce.Concurrency = 2
+	}
+
+	return p, nil
+}
+
+// Validate validates basic configuration of the project and returns an error if any of the settings contain illegal
+// values. This is not an exhaustive operation and further validation should be performed both in the client and/or
+// server side depending on the workflow that is executed.
+func Validate(p Project) error {
+	if p.Xcuit.App == "" {
+		return errors.New("missing path to app .ipa")
+	}
+	if !strings.HasSuffix(p.Xcuit.App, ".ipa") {
+		return fmt.Errorf("invaild application file: %s, make sure extension is .ipa", p.Xcuit.App)
+	}
+
+	if p.Xcuit.TestApp == "" {
+		return errors.New("missing path to test app .ipa")
+	}
+	if !strings.HasSuffix(p.Xcuit.TestApp, ".ipa") {
+		return fmt.Errorf("invaild application test file: %s, make sure extension is .ipa", p.Xcuit.TestApp)
+	}
+
+	if len(p.Suites) == 0 {
+		return errors.New("no suites defined")
+	}
+
+	for _, suite := range p.Suites {
+		if len(suite.Devices) == 0 {
+			return fmt.Errorf("missing devices configuration for suite: %s", suite.Name)
+		}
+		for didx, device := range suite.Devices {
+			if device.ID == "" && device.Name == "" {
+				return fmt.Errorf("missing device name or id for suite: %s. Devices index: %d", suite.Name, didx)
+			}
+
+			if strings.ToLower(device.PlatformName) != "ios" {
+				return fmt.Errorf("device platformName is incorrect for suite: %s. Devices index: %d. Supported device platform: iOS",
+					suite.Name, didx)
+			}
```

Comment:
This check is not required per se.  May be we should do like Espresso implementation, remove this check + [hardcode it](https://github.com/saucelabs/saucectl/blob/master/internal/espresso/config.go#L72) since it will be iOS.

It remove a potential error to the user for a setting that brings no value.

Reply:
I removed this check and set iOS as a default value instead.